RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -109,6 +114,29 @@ struct PurchaseButton: View {
 
 }
 
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+@available(tvOS, unavailable)
+private extension PurchaseButton {
+
+    var packagesProduceDifferentLabels: Bool {
+        return Set(
+            self.packages.all
+                .lazy
+                .map {
+                    IntroEligibilityStateView.text(
+                        for: .callToAction,
+                        localization: $0.localization,
+                        introEligibility: self.introEligibilityViewModel.allEligibility[$0.content]
+                    )
+                }
+        )
+        .count > 1
```

Comment:
This means we need to process the intro eligibility text for all packages in order to calculate the value (in android we would early exit the moment we find one that is different, but I think it's totally fine and probably cleaner 👍

Reply:
Oh yeah forgot to mention. Technically your implementation is more efficient, but we’re talking about <10 iterations if that, and each of those is just a few if statements.